linkerd/app/src/spire.rs

```diff
@@ -66,3 +79,21 @@ impl tower::Service<()> for Client {
         })
     }
 }
+
+#[cfg(not(target_os = "linux"))]
+impl tower::Service<()> for Client {
+    type Response = tonic::Response<watch::Receiver<client::SvidUpdate>>;
+    type Error = Error;
+    type Future = futures::future::BoxFuture<'static, Result<Self::Response, Self::Error>>;
+
+    fn poll_ready(
+        &mut self,
+        _cx: &mut std::task::Context<'_>,
+    ) -> std::task::Poll<Result<(), Self::Error>> {
+        unimplemented!("Spire is supported on Linux only")
+    }
+
+    fn call(&mut self, _req: ()) -> Self::Future {
+        unimplemented!("Spire is supported on Linux only")
+    }
+}
```

Comment:
this makes me a little nervous. why not avoid a service implementation entirely for non-linux builds, and handle this platform specific behavior at the call site where we use this client?

similarly, for the `From<Config>` implementation above, i'm worried this introduces the risk of runtime panics.

Reply:
AIUI, this is a first step that will setup the named pipe PR. Just a temporary landing spot.

Reply:
ah, good point. i'll mark this as resolved. :white_check_mark:

Reply:
Yes, my desire here is to isolate changes as much as possible to this module. And it is a temporary placeholder change to get us to compile for a windows target. I will follow with a named pipe support shortly after.